framework/contracts/account/src/contract.rs

```diff
@@ -182,19 +197,41 @@ pub fn instantiate(
     }
 
     if !install_modules.is_empty() {
+        let simulate_resp: SimulateInstallModulesResponse = deps.querier.query_wasm_smart(
+            config.module_factory_address.to_string(),
+            &abstract_std::module_factory::QueryMsg::SimulateInstallModules {
+                modules: install_modules.iter().map(|m| m.module.clone()).collect(),
+            },
+        )?;
+
+        simulate_resp.total_required_funds.iter().for_each(|funds| {
+            total_fee.add(funds.clone()).unwrap();
+        });
+
         // Install modules
         let (install_msgs, install_attribute) = _install_modules(
             deps.branch(),
             install_modules,
             config.module_factory_address,
-            config.version_control_address,
-            info.funds,
+            config.version_control_address.clone(),
+            simulate_resp.total_required_funds,
         )?;
         response = response
             .add_submessages(install_msgs)
             .add_attribute(install_attribute.key, install_attribute.value);
     }
 
+    for fee in total_fee.clone() {
+        let mut total_received = Coins::try_from(info.funds.clone()).unwrap();
+
+        total_received.sub(fee).map_err(|_| {
+            abstract_std::AbstractError::Fee(format!(
+                "Invalid fee payment sent. Expected {}, sent {:?}",
+                total_fee, info.funds
+            ))
+        })?;
+    }
```

Comment:
It looks incorrect(`total_received` created inside loop). This can allow to sub same fund multiple times. For example Carrot App and Dex Adapter both cost 5 juno to install, sending 5 juno will be enough to pass this loop

Reply:
Great catch, thanks Misha!